types/src/test_utils.rs

```diff
@@ -0,0 +1,449 @@
+    },
+    chain,
+    public_key::PublicKey,
+    Signature, Time,
+};
+
+use crate::block::{CommitExt, GENESIS_HEIGHT};
+use crate::consts::version;
+use crate::hash::{Hash, HashExt};
+use crate::nmt::{NamespacedHash, NamespacedHashExt};
+use crate::{DataAvailabilityHeader, ExtendedHeader, ValidatorSet};
+
+#[derive(Debug, Clone)]
+pub struct ExtendedHeaderGenerator {
+    chain_id: chain::Id,
+    key: SigningKey,
+    current_header: Option<ExtendedHeader>,
+}
+
+/// [`ExtendedHeader`] generator for testing purposes.
+///
+/// **WARNING: ALL METHODS PANIC! DO NOT USE IT IN PRODUCTION!**
+impl ExtendedHeaderGenerator {
+    /// Creates new `ExtendedHeaderGenerator`.
+    pub fn new() -> ExtendedHeaderGenerator {
+        let chain_id: chain::Id = "private".try_into().unwrap();
+
+        let key = SigningKey::new(rand::thread_rng());
+
+        ExtendedHeaderGenerator {
+            chain_id,
+            key,
+            current_header: None,
+        }
+    }
+
+    /// Creates new `ExtendedHeaderGenerator` and skip an amount of headers.
+    ///
+    /// This is equivalent to:
+    ///
+    /// ```ignore
+    /// let mut gen = ExtendedHeaderGenerator::new();
+    /// gen.skip(amount);
+    /// ```
+    pub fn new_skipped(amount: u64) -> ExtendedHeaderGenerator {
+        let mut gen = ExtendedHeaderGenerator::new();
+        gen.skip(amount);
+        gen
+    }
+
+    /// Returns the current header.
+    pub fn current_header(&self) -> &ExtendedHeader {
+        self.current_header
+            .as_ref()
+            .expect("genesis not constructed yet")
+    }
+
+    /// Returns the current header height.
+    pub fn current_height(&self) -> u64 {
+        self.current_header().height().value()
+    }
+
+    /// Generates the next header.
+    #[allow(clippy::should_implement_trait)]
+    pub fn next(&mut self) -> ExtendedHeader {
+        let header = match self.current_header {
+            Some(ref header) => generate_next(header, &self.key),
+            None => generate_genesis(&self.chain_id, &self.key),
+        };
+
+        self.current_header = Some(header.clone());
+        header
+    }
+
+    /// Generate the next amount of headers.
+    pub fn next_many(&mut self, amount: u64) -> Vec<ExtendedHeader> {
+        let mut headers = Vec::with_capacity(amount as usize);
+
+        for _ in 0..amount {
+            headers.push(self.next());
+        }
+
+        headers
+    }
+
+    /// Generates the next header of the provided header.
+    ///
+    /// This can be used to create two headers of same height but different hash.
+    ///
+    /// ```ignore
+    /// let mut gen = ExtendedHeaderGenerator::new();
+    /// let header1 = gen.next();
+    /// let header2 = gen.next();
+    /// let another_header2 = gen.next_of(&header1);
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// This method does not change the state of `ExtendedHeaderGenerator`.
+    pub fn next_of(&self, header: &ExtendedHeader) -> ExtendedHeader {
```

Comment:
as this doesn't advance the generator I'd name it `another_of`

Reply:
`another_of` would be nice if we provided the header with height 5 and generates another with height 5. In our case it will create header with height 6. So `next` needs to be in the name. Do you have any other suggestion?

Reply:
implemented `another_of` too